### PR TITLE
pass cargo args to idl generation

### LINKED ISSUE #3054: Compilation Error with Passing Cargo args when build program when IDL Generation is enabled in Anchor 0.30
Hello! I'm from [Debridge team](https://debridge.finance/). We faced an issue while updating our contract from anchor 0.28 to 0.30

### Description:
After updating from Anchor 0.28 to 0.30, our project encounters a compilation error when conditional compilation features are used along with IDL generation. The error does not occur when compiling with the --no-idl flag.

### MRE:

```rust
use anchor_lang::prelude::*;

declare_id!("8bBe1LzQ9cBVzk4bMqcYsRqM942C4DvYvwKy9bZqUVyg");

#[program]
pub mod mre_idl_bug {
    use cfg_if::cfg_if;
    use super::*;

    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
        cfg_if!(
           if #[cfg(feature = "one")] {
                const FOO: usize = 1;
           } else if #[cfg(feature = "two")] {
                const FOO: usize = 2;
           } else {
                compile_error!("You should use one of feature: `one`, `two`");
           }
        );

        msg!("Greetings from: {:?}", ctx.program_id);
        Ok(())
    }
}

#[derive(Accounts)]
pub struct Initialize {}
```

features "one" and "two" are added to Cargo.toml

### Expected behavior:

Running `anchor build -- --features one` should produce a valid library and an IDL.

### Actual result:
```
error: You should use one of feature: `one`, `two`
  --> programs/mre-idl-bug/src/lib.rs:17:17
   |
17 |                 compile_error!("You should use one of feature: `one`, `two`");
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `mre-idl-bug` (lib test) due to 1 previous error
Error: Building IDL failed
```

### Analysis:

The issue likely stems from the IDL builder function not accepting and passing cargo_args, based on the examination of the source code [here](https://github.com/coral-xyz/anchor/blob/70d4f488c1e8d759d337efe02546b81819d09fef/idl/src/build.rs#L70-L80).

It works fine when I add flag --no-idl.
It works fine with 0.28.0

### PR DESCRIPTION
**Pull request by biryukovmaxim**
closes #3054

### CODE REVIEW

File: idl/src/build.rs [hunk 1]
```diff
@@ -59,25 +66,30 @@ pub fn build_idl(
...
-
+    let mut args = vec![
+        &toolchain,
+        "test",
+        "__anchor_private_print_idl",
+        "--features",
+        "idl-build",
+    ];
+    args.extend(cargo_args.iter().map(String::as_str));
+    args.extend(["--", "--show-output", "--quiet"]);
```
Comment (acheroncrypto) [inline idl/src/build.rs:90]: Nit: [`args()`](https://doc.rust-lang.org/std/process/struct.Command.html#method.args) method extends the previous arguments on each call, so no need to declare separately and manually extend.

```rs
Command::new("cargo")
        .args([
            &toolchain,
            "test",
            "__anchor_private_print_idl",
            "--features",
            "idl-build",
        ])
        .args(cargo_args)
        .args(["--", "--show-output", "--quiet"])
```

**Review by acheroncrypto — CHANGES_REQUESTED** [review]
Thanks, it looks great other than the breaking change (new parameter).

**Review by acheroncrypto — COMMENTED** [review]
Thanks! One last thing before we merge, could you note this feature in the [CHANGELOG](https://github.com/coral-xyz/anchor/blob/5f1f72c511102ad49b0f4b9361efd714b6187e13/CHANGELOG.md#features)?

**Comment by biryukovmaxim** [comment]
> Thanks! One last thing before we merge, could you note this feature in the [CHANGELOG](https://github.com/coral-xyz/anchor/blob/5f1f72c511102ad49b0f4b9361efd714b6187e13/CHANGELOG.md#features)?

[done](https://github.com/coral-xyz/anchor/pull/3059/commits/1eabcdc2890dea42d6816b84da49730bb045db9f)

**Review by acheroncrypto — APPROVED** [review]
Thank you!

**Comment by Henry-E** [comment]
closes: 1987

**Pull request merged**